src/main/java/de/tum/in/www1/artemis/domain/participation/Participation.java

```diff
@@ -230,7 +230,7 @@ private Result findLatestResult(boolean filterIllegalResults) {
         }
 
         List<Result> sortedResultsWithCompletionDate = results.stream().filter(r -> r.getCompletionDate() != null)
-                .sorted((r1, r2) -> r2.getCompletionDate().compareTo(r1.getCompletionDate())).collect(Collectors.toList());
+                .sorted((r1, r2) -> r2.getCompletionDate().compareTo(r1.getCompletionDate())).toList();
```

Comment:
```suggestion
                .sorted(Comparator.comparing(Result::getCompletionDate).reversed()).toList();
```
Maybe that's better, up to you.

Reply:
Integrated. I reversed() the comparator because the old code uses r2 as the first argument, Comparator.comparing() calls r1.compareTo(r2).